pkg/sql/authorization.go

```diff
@@ -142,13 +158,22 @@ func (p *planner) CheckAnyPrivilege(ctx context.Context, descriptor sqlbase.Desc
 }
 
 // HasAdminRole implements the AuthorizationAccessor interface.
+// Requires a valid transaction to be open.
 func (p *planner) HasAdminRole(ctx context.Context) (bool, error) {
 	user := p.SessionData().User
 	if user == "" {
 		return false, errors.AssertionFailedf("empty user")
 	}
+	// Verify that the txn is valid in any case, so that
+	// we don't get the risk to say "OK" to root requests
+	// with an invalid API usage.
+	if p.txn == nil || !p.txn.IsOpen() {
+		return false, errors.AssertionFailedf("cannot use HasAdminRole without a txn")
+	}
 
 	// Check if user is 'root' or 'node'.
+	// TODO(knz): planner HasAdminRole has no business authorizing
```

Comment:
nit: I don't agree with this TODO (migrations or jobs or other internal processes that aren't on behalf of a user session?) but don't let that block merging.

Reply:
Discussed with @aaron-crl previously: we should not let the `node` principal do anything else than node-node RPCs.

The "internal" SQL activity should still execute as `root`, or some new principal which we'd dedicate to that purpose.